services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/DynamoDbAsyncTable.java

```diff
@@ -212,6 +212,28 @@
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * Deletes a single item from the mapped table using just the key of a supplied modelled 'key item' object.
```

Comment:
Should we emphasize that the item you're returning isn't necessarily the one you send in?